pkg/operator/resource/resourceapply/admissionregistration.go

```diff
@@ -164,3 +166,121 @@ func copyValidatingWebhookCABundle(from, to *admissionregistrationv1.ValidatingW
 		}
 	}
 }
+
+// ApplyValidatingAdmissionPolicyV1beta1 ensures the form of the specified
+// validatingadmissionpolicyconfiguration is present in the API. If it does not exist,
+// it will be created. If it does exist, the metadata of the required
+// validatingadmissionpolicyconfiguration will be merged with the existing validatingadmissionpolicyconfiguration
+// and an update performed if the validatingadmissionpolicyconfiguration spec and metadata differ from
+// the previously required spec and metadata based on generation change.
+func ApplyValidatingAdmissionPolicyV1beta1(ctx context.Context, client admissionregistrationclientv1beta1.ValidatingAdmissionPoliciesGetter, recorder events.Recorder,
+	requiredOriginal *admissionregistrationv1beta1.ValidatingAdmissionPolicy, cache ResourceCache) (*admissionregistrationv1beta1.ValidatingAdmissionPolicy, bool, error) {
+	if requiredOriginal == nil {
+		return nil, false, fmt.Errorf("Unexpected nil instead of an object")
+	}
+
+	existing, err := client.ValidatingAdmissionPolicies().Get(ctx, requiredOriginal.GetName(), metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		required := requiredOriginal.DeepCopy()
+		actual, err := client.ValidatingAdmissionPolicies().Create(
+			ctx, resourcemerge.WithCleanLabelsAndAnnotations(required).(*admissionregistrationv1beta1.ValidatingAdmissionPolicy), metav1.CreateOptions{})
+		reportCreateEvent(recorder, required, err)
+		if err != nil {
+			return nil, false, err
+		}
+		// need to store the original so that the early comparison of hashes is done based on the original, not a mutated copy
+		cache.UpdateCachedResourceMetadata(requiredOriginal, actual)
+		return actual, true, nil
+	} else if err != nil {
+		return nil, false, err
+	}
+
+	if cache.SafeToSkipApply(requiredOriginal, existing) {
+		return existing, false, nil
+	}
+
+	required := requiredOriginal.DeepCopy()
+	modified := resourcemerge.BoolPtr(false)
```

Comment:
This one should be deprecated soon: https://github.com/openshift/library-go/pull/1643/files#diff-9ad12a3ee862cb99a685d9fdaafdce7fb3b767eef8927b49025de62506b6fe95R73